src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -508,6 +518,8 @@ private static ArgumentInfo GetArgumentInfo(Type type, object defaults)
 
                     // Note that we put the default arg in the map to ensure that no other args use the same name.
                     map.Add(name.ToLowerInvariant(), arg);
+                    if (attrName != null)
+                        map.Add(attrName.ToLowerInvariant(), arg);
```

Comment:
>ToLowerInvariant [](start = 41, length = 16)

If you can do a case insensitive comparer for the map the toLowerInvariant is not needed. However if you want to convert to lower invariant for other reasons, I would suggest doing the conversion at the top of the function and using that through the rest of the code. That way you do the conversion once.